src/test/java/com/google/api/generator/engine/ast/JavaDocCommentTest.java

```diff
@@ -0,0 +1,150 @@
+    String content = "this is a test comment";
+    JavaDocComment javaDocComment = JavaDocComment.withComment(content);
+    assertEquals(javaDocComment.comment(), content);
+  }
+
+  @Test
+  public void createJavaDocComment_multipleComments() {
+    String comment1 = "This is a test comment.";
+    String comment2 = "This is unordered list.";
+    String comment3 = "This is ordered list.";
+    List<String> orderedList =
+        Arrays.asList("A flattened method.", "A request object method.", "A callable method.");
+    List<String> unOrderedList =
+        Arrays.asList("A flattened method.", "A request object method.", "A callable method.");
+    JavaDocComment javaDocComment =
+        JavaDocComment.builder()
+            .addComment(comment1)
+            .addComment(comment2)
+            .addUnorderedList(unOrderedList)
+            .addComment(comment3)
+            .addOrderedList(orderedList)
+            .build();
+    String expected =
+        "This is a test comment.\n"
+            + "This is unordered list.\n"
+            + "<ul>\n"
+            + "<li> A flattened method.\n"
+            + "<li> A request object method.\n"
+            + "<li> A callable method.\n"
+            + "</ul>\n"
+            + "This is ordered list.\n"
+            + "<ol>\n"
+            + "<li> A flattened method.\n"
+            + "<li> A request object method.\n"
+            + "<li> A callable method.\n"
+            + "</ol>";
+    assertEquals(javaDocComment.comment(), expected);
+  }
+
+  @Test
+  public void createJavaDocComment_multipleParams() {
+    String paramName1 = "shelfName";
+    String paramDescription1 = "The name of the shelf where books are published to.";
+    String paramName2 = "shelfId";
+    String paramDescription2 = "The shelfId of the shelf where books are published to.";
+    JavaDocComment javaDocComment =
+        JavaDocComment.builder()
+            .addParam(paramName1, paramDescription1)
+            .addParam(paramName2, paramDescription2)
+            .build();
+    String expected =
+        "@param shelfName The name of the shelf where books are published to.\n"
+            + "@param shelfId The shelfId of the shelf where books are published to.";
+    assertEquals(javaDocComment.comment(), expected);
+  }
+
+  @Test
+  public void createJavaDocComment_sampleCode() {
+    String comment = "sample codes:";
+    StringObjectValue stringObjectValue =
+        StringObjectValue.withValue("project/{project}/shelfId/{shelfId}");
+    ValueExpr valueExpr = ValueExpr.builder().setValue(stringObjectValue).build();
+    VariableExpr expr =
+        VariableExpr.builder()
+            .setVariable(Variable.builder().setName("resource").setType(TypeNode.STRING).build())
+            .build();
+    AssignmentExpr assignmentExpr =
+        AssignmentExpr.builder().setVariableExpr(expr).setValueExpr(valueExpr).build();
+    JavaWriterVisitor javaWriterVisitor = new JavaWriterVisitor();
```

Comment:
Why is the writerVisitor in this test? It should not be needed, could we please remove it?

Reply:
this is for writing the assignmentExpr which is used to built sample code. But since we are just testing JavaDocComment related code, I changed the sample code to a plain string.